fe/src/main/java/org/apache/doris/system/SystemInfoService.java

```diff
@@ -729,8 +639,18 @@ public void releaseBackends(String clusterName, boolean isReplay) {
     // return null if not enough backend
     // use synchronized to run serially
     public synchronized List<Long> seqChooseBackendIds(int backendNum, boolean needAlive, boolean isCreate,
-            String clusterName) {
-        long lastBackendId = -1L;
+                                                       String clusterName) {
+        return seqChooseBackendIds(backendNum, needAlive, isCreate, clusterName, getClusterBackends(clusterName));
+    }
+    public synchronized List<Long> seqChooseBackendIdsByStorageMedium(int backendNum, boolean needAlive, boolean isCreate,
+                                                                      String clusterName, TStorageMedium storageMedium) {
+        final List<Backend> backends = getClusterBackends(clusterName).stream().filter(v -> !v.checkDiskExceedLimitByStorageMedium(storageMedium)).collect(Collectors.toList());
```

Comment:
This is not user friendly. For example, if all backends not meet the `checkDiskExceedLimitByStorageMedium()`, the user will only get the msg: "Not enough backends", but he will not know why there is no enough backend.
So could you please find a way to return the detail msg to user about the reason of failure?

Reply:
actually, I change the detail msg to `throw new DdlException("Failed to find enough host in all backends with storage medium is " + storageMedium + ". need: " + replicationNum`